api/replica.go

```diff
@@ -14,20 +15,37 @@ type Replica struct {
 	handler   http.Handler
 	tlsConfig *tls.Config
 	primary   string
+	addr      string // the address of this replica
+	mu        sync.RWMutex
```

Comment:
We don't seem to be using the `RLock()` method anywhere; is there a reason we need the `RWMutex` instead of a regular mutex?

Reply:
We don't call the method directly, but we pass the output of `mu.RLocker` as the argument to `NewCond`, which means that the lock acquired and released by the Cond is a read lock. So down in `ServeHTTP`, we're actually acquiring read locks, despite saying `Lock` and `Unlock`.

Reply:
Ah okay, that makes sense. In that case, let's add a comment above line 42 and line 65 indicating that we're getting a read lock and a write lock, respectively.